internal/prune/agent/execute.go

```diff
@@ -0,0 +1,66 @@
+/*
+ *  *******************************************************************************
+ *  * Copyright (c) 2019 Edgeworx, Inc.
+ *  *
+ *  * This program and the accompanying materials are made available under the
+ *  * terms of the Eclipse Public License v. 2.0 which is available at
+ *  * http://www.eclipse.org/legal/epl-2.0
+ *  *
+ *  * SPDX-License-Identifier: EPL-2.0
+ *  *******************************************************************************
+ *
+ */
+
+package pruneagent
+
+import (
+	"github.com/eclipse-iofog/iofogctl/internal/config"
+	"github.com/eclipse-iofog/iofogctl/internal/execute"
+	"github.com/eclipse-iofog/iofogctl/pkg/util"
+)
+
+type executor struct {
+	name        string
+	namespace   string
+	useDetached bool
+}
+
+func NewExecutor(namespace, name string, useDetached bool) (execute.Executor, error) {
+	return executor{name: name, namespace: namespace, useDetached: useDetached}, nil
+}
+
+func (exe executor) GetName() string {
+	return exe.name
+}
+
+func (exe executor) Execute() error {
+	util.SpinStart("Pruning Agent")
+	// Get Agent from config
+	var agent config.Agent
+	var err error
+	if exe.useDetached {
+		agent, err = config.GetDetachedAgent(exe.name)
+	} else {
+		agent, err = config.GetAgent(exe.namespace, exe.name)
+	}
+
+	if err == nil {
+		// Deprovision agent
```

Comment:
Same here. It does not deprovision agent

Reply:
updated